packages/backend/src/services/ProjectService/ProjectService.ts

```diff
@@ -111,79 +90,94 @@ export class ProjectService {
-        );
+                const explores = await this.jobModel.tryJobStep(
+                    job.jobUuid,
+                    JobStepType.COMPILING,
+                    async () => adapter.compileAllExplores(),
+                );
 
-        await this.jobModel.tryJobStep(
-            jobUuid,
-            JobStepType.CACHING,
-            async () => {
+                const projectUuid = await this.jobModel.tryJobStep(
+                    job.jobUuid,
+                    JobStepType.CREATING_PROJECT,
+                    async () =>
+                        this.projectModel.create(user.organizationUuid, data),
+                );
                 await this.projectModel.saveExploresToCache(
                     projectUuid,
                     explores,
                 );
-            },
-        );
 
-        await this.jobModel.upsertJobStatus(
-            jobUuid,
-            projectUuid,
-            JobStatusType.DONE,
-        );
+                // do we still need project adapters?
+                this.projectAdapters[projectUuid] = adapter;
 
-        analytics.track({
-            event: 'project.created',
-            userId: user.userUuid,
-            properties: {
-                projectName: data.name,
-                projectId: projectUuid,
-                projectType: data.dbtConnection.type,
-                warehouseConnectionType: data.warehouseConnection.type,
-                organizationId: user.organizationUuid,
-                dbtConnectionType: data.dbtConnection.type,
-            },
-        });
-        this.projectLoading[projectUuid] = false;
-        this.projectAdapters[projectUuid] = adapter;
-        return this.getProject(projectUuid, user);
+                await this.jobModel.update(job.jobUuid, {
+                    jobStatus: JobStatusType.DONE,
+                    jobResults: {
+                        projectUuid,
+                    },
+                });
+                analytics.track({
+                    event: 'project.created',
+                    userId: user.userUuid,
+                    properties: {
+                        projectName: data.name,
+                        projectId: projectUuid,
+                        projectType: data.dbtConnection.type,
+                        warehouseConnectionType: data.warehouseConnection.type,
+                        organizationId: user.organizationUuid,
+                        dbtConnectionType: data.dbtConnection.type,
+                    },
+                });
+                this.projectAdapters[projectUuid] = adapter;
+            } catch (error) {
+                await this.jobModel.setPendingJobsToSkipped(job.jobUuid);
+                throw error;
+            }
+        };
+
+        await this.jobModel.create(job);
+        doAsyncWork()
+            .catch(async (e) => {
+                await this.jobModel.update(job.jobUuid, {
```

Comment:
Do we need to call `await this.jobModel.setPendingJobsToSkipped(job.jobUuid);` as we do on create.